Scripts/World/Chunk.cs

```diff
@@ -71,18 +73,21 @@ public override void _Process(float delta)
             meshInstance.SetMesh(mesh);
         }
     }
-
     public void UpdateMesh()
     {
-        System.Threading.Thread thread = new System.Threading.Thread(() =>
+        if(generationThread != null && generationThread.ThreadState == System.Threading.ThreadState.Running)
+            generationThread.Abort();
+        
+        generationThread = new System.Threading.Thread(() =>
         {
             GenerateSurface();
             generationFinished = true;
         })
         {
-            Priority = System.Threading.ThreadPriority.Lowest
+            Priority = System.Threading.ThreadPriority.Highest
```

Comment:
why?

Reply:
The lag was caused by the overhead of Godot things which have to be in main thread, so I thought its best to get the generation done asap, since it doesn't hold up the player moving or anything